examples/plot_metadata_routing.py

```diff
@@ -0,0 +1,615 @@
+        # and then prepare the input to the underlying ``predict`` method.
+        params = request_router.get_params(params=predict_params, method="predict")
+        return self.estimator_.predict(X, **params.estimator.predict)
+
+    def get_metadata_routing(self):
+        router = (
+            MetadataRouter()
+            .add_self(self)
+            .add(estimator=self.estimator, method_mapping="one-to-one")
+        )
+        return router.serialize()
+
+
+# %%
+# The key parts where the above estimator differs from our previous
+# meta-estimator is accepting ``sample_weight`` explicitly in ``fit`` and
+# including it in ``fit_params``. Making ``sample_weight`` an explicit argument
+# makes sure ``fit_requests(sample_weight=...)`` is present for this class.
+#
+# In ``get_metadata_routing``, we add ``self`` to the routing using
+# ``add_self``. Now let's look at some examples:
+
+# %%
+# no metadata requested
+est = RouterConsumerClassifier(estimator=ExampleClassifier())
+est.get_metadata_routing()
+
+
+# %%
+# ``sample_weight`` requested by child estimator
+est = RouterConsumerClassifier(
+    estimator=ExampleClassifier().fit_requests(sample_weight=True)
+)
+est.get_metadata_routing()
+# %%
+# ``sample_weight`` requested by meta-estimator
+est = RouterConsumerClassifier(estimator=ExampleClassifier()).fit_requests(
+    sample_weight=True
+)
+est.get_metadata_routing()
+
+# %%
+# Note the difference in the requested meatada representations above.
+#
+# We can also alias the metadata to pass different values to them:
+
+est = RouterConsumerClassifier(
+    estimator=ExampleClassifier().fit_requests(sample_weight="clf_sample_weight"),
+).fit_requests(sample_weight="meta_clf_sample_weight")
+est.get_metadata_routing()
+
+# %%
+# However, ``fit`` of the meta-estimator only needs the alias for the
+# sub-estimator, since it doesn't validate and route its own required metadata:
+est.fit(X, y, sample_weight=my_weights, clf_sample_weight=my_other_weights)
+
+# %%
+# Alias only on the sub-estimator. This is useful if we don't want the
+# meta-estimator to use the metadata, and we only want the metadata to be used
+# by the sub-estimator.
+est = RouterConsumerClassifier(
+    estimator=ExampleClassifier().fit_requests(sample_weight="aliased_sample_weight")
+).fit_requests(sample_weight=True)
+est.get_metadata_routing()
+
+
+# %%
+# Simple Pipeline
+# ---------------
+# A slightly more complicated use-case is a meta-estimator which does something
+# similar to the ``Pipeline``. Here is a meta-estimator, which accepts a
+# transformer and a classifier, and applies the transformer before running the
+# classifier.
+
+
+class SimplePipeline(ClassifierMixin, BaseEstimator):
+    _required_parameters = ["estimator"]
+
+    def __init__(self, transformer, classifier):
+        self.transformer = transformer
+        self.classifier = classifier
+
+    @process_routing
+    def fit(self, X, y, **fit_params):
```

Comment:
Is it appropriate to actually make this:
```
    def fit(self, X, y, routed_params):
```
??

I guess this is a bit magical for a reader of the code.